src/models/Country.php

```diff
@@ -123,91 +125,107 @@ function($a, $b) {
+    if (!$mc_result || count($mc_result) === 0 || $refresh) {
+      $all_countries =
+        await self::genAll('SELECT * FROM countries WHERE enabled = 1');
+      self::setMCRecords('ALL_ENABLED_COUNTRIES', $all_countries);
+    }
+    /* HH_IGNORE_ERROR[4110]: getMCRecords returns a 'mixed' type, HHVM is unsure of the type at this point */
+    return self::getMCRecords('ALL_ENABLED_COUNTRIES');
   }
 
   // All enabled countries. The weird sorting is because SVG lack of z-index
   // and things looking like shit in the map. See issue #20.
   public static async function genAllEnabledCountriesForMap(
+    bool $refresh = false,
   ): Awaitable<array<Country>> {
-    return await self::genAll(
-      'SELECT * FROM countries WHERE enabled = 1 ORDER BY CHAR_LENGTH(d)',
-      self::MC_KEY_ALL_ENABLED_COUNTRIES_FOR_MAP,
-    );
+    $mc_result = self::getMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP');
+    if (!$mc_result || count($mc_result) === 0 || $refresh) {
+      $all_countries = await self::genAll(
+        'SELECT * FROM countries WHERE enabled = 1 ORDER BY CHAR_LENGTH(d)',
+      );
+      self::setMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP', $all_countries);
+    }
+    /* HH_IGNORE_ERROR[4110]: getMCRecords returns a 'mixed' type, HHVM is unsure of the type at this point */
+    return self::getMCRecords('ALL_ENABLED_COUNTRIES_FOR_MAP');
   }
 
   // All enabled and unused countries
   public static async function genAllAvailableCountries(
+    bool $refresh = false,
   ): Awaitable<array<Country>> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT * FROM countries WHERE enabled = 1 AND used = 0',
-    );
-
-    $countries = array();
-    foreach ($result->mapRows() as $row) {
-      $countries[] = await self::countryFromRow($row->toArray());
+    $mc_result = self::getMCRecords('ALL_AVAILABLE_COUNTRIES');
+    if (!$mc_result || count($mc_result) === 0 || $refresh) {
+      $all_countries = await self::genAll(
+        'SELECT * FROM countries WHERE enabled = 1 AND used = 0',
+      );
+      self::setMCRecords('ALL_AVAILABLE_COUNTRIES', $all_countries);
     }
-
-    usort(
-      $countries,
-      function($a, $b) {
-        return strcmp($a->name, $b->name);
-      },
-    );
-
-    return $countries;
+    /* HH_IGNORE_ERROR[4110]: getMCRecords returns a 'mixed' type, HHVM is unsure of the type at this point */
+    return self::getMCRecords('ALL_AVAILABLE_COUNTRIES');
   }
 
   // Check if country is in an active level
   public static async function genIsActiveLevel(
     int $country_id,
   ): Awaitable<bool> {
-    $db = await self::genDb();
-
-    $result = await $db->queryf(
-      'SELECT COUNT(*) FROM levels WHERE entity_id = %d AND active = 1',
-      $country_id,
-    );
-
-    invariant($result->numRows() === 1, 'Expected exactly one result');
-    return intval(firstx($result->mapRows())['COUNT(*)']) > 0;
+    return Level::genWhoUses($country_id) != null;
```

Comment:
Level::genWhoUses() returns a Level based on the input of the country Id, or _null_ otherwise.  Country::genIsActiveLevel(), by supplying a country Id, is used to determine if a country is in use.  So in this case, if Level::genWhoUses($country_id) returns _null_ the country is not in use, whereas, if the method returns a Level (_not null_) the country is in use.